src/pymor/discretizers/builtin/cg.py

```diff
@@ -1067,39 +1073,62 @@ def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretiz
         BoundaryL2Functional = BoundaryL2ProductFunctional
 
     Li = [DiffusionOperator(grid, boundary_info, diffusion_constant=0, name='boundary_part')]
+    if energy_product:
+        eLi = [DiffusionOperator(grid, boundary_info, dirichlet_clear_columns=True, diffusion_constant=0)]
     coefficients = [1.]
 
     # diffusion part
     if isinstance(p.diffusion, LincombFunction):
         Li += [DiffusionOperator(grid, boundary_info, diffusion_function=df, dirichlet_clear_diag=True,
                                  name=f'diffusion_{i}')
                for i, df in enumerate(p.diffusion.functions)]
+        if energy_product:
+            eLi += [DiffusionOperator(grid, boundary_info, diffusion_function=df, dirichlet_clear_diag=True,
+                                      dirichlet_clear_columns=True)
+                   for i, df in enumerate(p.diffusion.functions)]
```

Comment:
Since you only want to assemble for a fixed parameter, it does not make sense to iterate over the coefficients. Just instantiate `DiffusionOperator` with `p.diffusion`.